src/code/z_parameter.c

```diff
@@ -1145,26 +1149,484 @@ void Inventory_ChangeAmmo(s16 item, s16 ammoChange) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Parameter_AddMagic.s")
+void Magic_Add(PlayState* play, s16 magicToAdd) {
+    if (((void)0, gSaveContext.save.playerData.magic) < ((void)0, gSaveContext.magicCapacity)) {
+        gSaveContext.magicToAdd += magicToAdd;
+        gSaveContext.isMagicRequested = true;
+    }
+}
+
+void Magic_Reset(PlayState* play) {
+    if ((gSaveContext.magicState != MAGIC_STATE_STEP_CAPACITY) && (gSaveContext.magicState != MAGIC_STATE_FILL)) {
+        sMagicMeterOutlinePrimRed = sMagicMeterOutlinePrimGreen = sMagicMeterOutlinePrimBlue = 255;
+        gSaveContext.magicState = MAGIC_STATE_IDLE;
+    }
+}
+
+/**
+ * Request to consume magic.
+ *
+ * @param magicToConsume the positive-valued amount to decrease magic by
+ * @param type how the magic is consumed.
+ * @return false if the request failed
+ */
+s32 Magic_Consume(PlayState* play, s16 magicToConsume, s16 type) {
+    InterfaceContext* interfaceCtx = &play->interfaceCtx;
+
+    // Magic is not acquired yet
+    if (!gSaveContext.save.playerData.isMagicAcquired) {
+        return false;
+    }
+
+    // Not enough magic available to consume
+    if ((gSaveContext.save.playerData.magic - magicToConsume) < 0) {
+        if (gSaveContext.magicCapacity != 0) {
+            play_sound(NA_SE_SY_ERROR);
+        }
+        return false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115D5C.s")
+    switch (type) {
+        case MAGIC_CONSUME_NOW:
+        case MAGIC_CONSUME_NOW_ALT:
+            // Drank magic immediately
+            // Ex. Deku Bubble
```

Comment:
```suggestion
            // e.g. Deku Bubble
```
and not really any need to be on its own line imo.